wagtail/contrib/search_promotions/views.py

```diff
@@ -1,85 +1,64 @@
 from django.urls import reverse
 from django.utils.translation import gettext as _
-from django.views.decorators.vary import vary_on_headers
 
 from wagtail.admin import messages
-from wagtail.admin.auth import any_permission_required, permission_required
+from wagtail.admin.auth import permission_required
 from wagtail.admin.forms.search import SearchForm
 from wagtail.admin.modal_workflow import render_modal_workflow
+from wagtail.admin.ui.tables import Column, TitleColumn
+from wagtail.admin.views import generic
 from wagtail.contrib.search_promotions import forms, models
 from wagtail.contrib.search_promotions.models import Query
 from wagtail.log_actions import log
 from wagtail.search.utils import normalise_query_string
 
 
-@any_permission_required(
-    "wagtailsearchpromotions.add_searchpromotion",
-    "wagtailsearchpromotions.change_searchpromotion",
-    "wagtailsearchpromotions.delete_searchpromotion",
-)
-@vary_on_headers("X-Requested-With")
-def index(request):
-    # Ordering
-    valid_ordering = ["query_string", "-query_string", "views", "-views"]
-    ordering = valid_ordering[0]
-
-    if "ordering" in request.GET and request.GET["ordering"] in valid_ordering:
-        ordering = request.GET["ordering"]
-
-    # Query
-    queries = Query.objects.filter(editors_picks__isnull=False).distinct()
-
-    if "views" in ordering:
-        queries = queries.annotate(views=functions.Coalesce(Sum("daily_hits__hits"), 0))
-
-    queries = queries.order_by(ordering)
-
-    # Search
-    is_searching = False
-    query_string = request.GET.get("q", "")
-
-    if query_string:
-        queries = queries.filter(query_string__icontains=query_string)
-        is_searching = True
-
-    # Paginate
-    paginator = Paginator(queries, per_page=20)
-    try:
-        queries = paginator.page(request.GET.get("p", 1))
-    except InvalidPage:
-        raise Http404
-
-    if request.headers.get("x-requested-with") == "XMLHttpRequest":
-        return TemplateResponse(
-            request,
-            "wagtailsearchpromotions/results.html",
-            {
-                "is_searching": is_searching,
-                "ordering": ordering,
-                "queries": queries,
-                "query_string": query_string,
-            },
-        )
-    else:
-        return TemplateResponse(
-            request,
-            "wagtailsearchpromotions/index.html",
-            {
-                "is_searching": is_searching,
-                "ordering": ordering,
-                "queries": queries,
-                "query_string": query_string,
-                "search_form": SearchForm(
-                    data={"q": query_string} if query_string else None,
-                ),
-            },
+class IndexView(generic.IndexView):
+    model = Query
+    template_name = "wagtailsearchpromotions/index.html"
+    results_template_name = "wagtailsearchpromotions/results.html"
+    page_title = _("Search Terms")
```

Comment:
```suggestion
    page_title = gettext_lazy("Search Terms")
```

(and other translatable strings at class definition level)